dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/ShortestResponseLoadBalance.java

```diff
@@ -77,11 +120,22 @@
                 shortestIndexes[shortestCount++] = i;
                 totalWeight += afterWarmup;
                 if (sameWeight && i > 0
-                        && afterWarmup != firstWeight) {
+                    && afterWarmup != firstWeight) {
                     sameWeight = false;
                 }
             }
         }
+
+        if (System.currentTimeMillis() - lastUpdateTime > SLIDE_PERIOD
+            && onResetSlideWindow.compareAndSet(false, true)) {
+            //reset slideWindowData in async way
+            Executors.newSingleThreadExecutor().execute(() -> {
```

Comment:
will this create a lot of threads in a heavy load

Reply:
You are right. I will fix  the prolem later.